src/cpu/jit_uni_pooling.hpp

```diff
@@ -50,7 +53,7 @@ struct jit_uni_pooling_fwd_t: public cpu_primitive_t {
             bool ok = true
                 && implication(desc()->src_desc.ndims == 5,
                         isa == avx512_common)
-                && mayiuse(isa)
+                && (mayiuse(isa) || (isa == avx2 && mayiuse(avx)))
```

Comment:
I would suggest to left mayiuse(isa) here and add an instantiation of jit_uni_pooling_fwd_t<avx> to be consistent with other ISAs.
Places of the instantiations:
- https://github.com/intel/mkl-dnn/blob/master/src/cpu/jit_uni_pooling.cpp#L289
- https://github.com/intel/mkl-dnn/blob/master/src/cpu/cpu_engine.cpp#L195

Reply:
I don't completely agree with that. Mainly for 2 reasons:
1. we end up with 2 very similar instantiations (the differences would be really minor)
2. it is already not true, e.g. avx512_mic we optionally put prefetches, while the code is instatiated for avx512_f.

To me it makes more sense to replace avx2 with avx, and whenever it makes sense have a sequence of avx2 instructions under `if(avx2)` while keeping the avx path as a default.

That approach is already taken in e.g. avx jit gemm (that is not yet promoted here).